aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -108,6 +109,35 @@ namespace detail {
   }
 }
 
+/**
+ * This is used to register callback of specific key to PyTorch for
+ * computing DispatchKeySet additionally.
```

Comment:
Huh, I was NOT expecting this

Reply:
Don't do a callback. DispatchKey is fixed and not open registration. So you can just directly inline a test for autocast special case in the logic below.

Reply:
Actually, looking over your autocast callback, I am wondering if it would be a better idea to put the logic for testing the bit in the autocast kernels themselves. Yes, you have to remember to check if the backend is enabled for all those kernels, but there is a big benefit: no one has to pay for the special autocast behavior (which is what happens when you put it here: ALL ops pay, even if they're not using autocast at all.)

Reply:
What was the verdict here?